### DISCUSSION TITLE
深度终端

### DISCUSSION THREAD

**Discussion by ahvlee** [linuxdeepin/developer-center#4961] [Q&A | 问答板块] [answered]
你好。我在bbs.deepin.org已经问一下了，而且liwl1991告诉我需要来这里。
https://bbs.deepin.org/post/259214?id=259214&limit=20&offset=0&postId=1490149

我在学习。还是不明白太多东西。谢谢帮助。

**Comment by BLumia** [comment, marked as answer]
> 1。编辑以后我在README.zh_CN一样sudo make install。如果要删除安转的文件怎么办？ 

CMake 默认并不会生成 uninstall 目标，所以你如果选择这种方式，那你需要根据安装过程最后的输出记录所安装的文件位置，然后手动移除这些文件。

> 怎么可以建作debian的软件包？

简要答案是在源码根目录执行 `debuild -uc -us -nc`，当然如果你对此感兴趣，建议阅读 debian 的打包指南。（项目里已经附带了 debian 文件夹，包含了所需的打包配置）

> 2023-07-11, 14:18:36.027 [Debug ] [ ] The font to use in the terminal can not be matched exactly on your system.
> 2023-07-11, 14:18:36.028 [Debug ] [ ] Selected: "Noto Sans Mono,14,-1,2,50,0,0,0,0,0"
> 2023-07-11, 14:18:36.028 [Debug ] [ ] System : "Noto Sans Mono,14.25,-1,2,50,0,0,0,0,0"
>
> 这是什么意思呢？我在系统有安转的Noto字体。是错误吗？

如果实际生效的是 Noto Sans Mono 的话，可以忽略这个输出。顺便 cc @ArchieMeng 

> 在GUI里面无法变换快捷经。在.config/deepin-terminal/config.confl里面些...

界面没提供的话是设计上不希望你换的。你后者列出的是旧版终端所用的配置，新版本已经不是这个配置文件了。

> 如何有方法执行deepin-terminal时候，自动开在一样的终端里面开六个Tab吗？

目前没有（但这个需求好奇怪...）

**Reply by ahvlee** [reply]
为什么奇怪呢？
比如用gnome-terminal时候我执行一个小bash脚本。
#/bin/bash
/usr/bin/gnome-terminal.real --geometry 116x25-26-29 --tab --title="三文鱼" --command="fish" \
		--tab --title="ipython" --command "ipython3" \
		--tab --profile=solarized --title="工作" --active \
		--tab --profile=solarized --title="tmux" \
		--tab --profile=solarized --title="docu" \
		--tab -vv --profile=lee --title="执行" 2>>/tmp/mgt.log  2>&1
如何做这个用deepin-terminal?

**Reply by BLumia** [reply]
你最初的描述没说是打开多个终端并通过参数执行不同的命令，所以只是按存粹打开六个tab去理解的话确实...很奇怪。

关于这个 `--tab` 参数的特性请求应该有一些意义。 cc @meiyixiang

**Comment by ArchieMeng** [comment]
关于字体这个Log，我看我的也有。应该也不影响。

**Comment by ArchieMeng** [comment]
关于版本的问题，无法复现。能提供一下你编译的git版本吗？

**Reply by ahvlee** [reply]
是哪个命令？ 

$>git version
git version 2.37.2
$>git remote show origin
* 远程 origin
  获取地址：https://github.com/linuxdeepin/deepin-terminal
  推送地址：https://github.com/linuxdeepin/deepin-terminal
  HEAD 分支：master
  远程分支：
    community     已跟踪
    master        已跟踪
    qtermwidget   已跟踪
    release/1050  已跟踪
    release/eagle 已跟踪
    rpm           已跟踪
  为 'git pull' 配置的本地分支：
    master 与远程 master 合并
  为 'git push' 配置的本地引用：
    master 推送至 master (本地已过时)

$> git log
commit b9ef82d2561563031f6bbff95ae93779f210dacc (HEAD -> master, origin/master, origin/HEAD)
Author: asterwyx <yixue.wang@outlook.com>
Date:   Wed Jun 21 16:05:56 2023 +0800

    fix: compile error (#283)

    Miss using namespace Dtk::Gui;

    Log: fix compile error

commit d8dac88f1117c0f0fb9deb27638efd1fac70a13d
Author: ArchieMeng <mengyutao@deepin.org>
Date:   Mon Jun 12 14:23:11 2023 +0800

    feat: 将背景模糊设置项移动到界面组和透明度放在一起 (#278)

    Issue: https://github.com/linuxdeepin/developer-center/issues/4379
    Log: 将背景模糊设置项移动到界面组和透明度放在一起

    Signed-off-by: Yutao Meng <mengyutao@deepin.org>

commit 7f1691bcc681c722748efc5e4da0e849230be132
Author: ArchieMeng <mengyutao@deepin.org>
Date:   Tue May 16 13:20:21 2023 +0800

**Reply by ArchieMeng** [reply]
我想起来了，你如果是跑的直接cmake编译出来的，而不是dpkg打包的版本，那版本显示就是空的。终端的版本信息是靠cmake配置时候的VERSION变量指定的。

**Comment by meiyixiang** [comment]
我有些不理解看起来都在描述技术问题！
/usr/bin/gnome-terminal.real --geometry 116x25-26-29 --tab --title="三文鱼" --command="fish"
--tab --title="ipython" --command "ipython3"
--tab --profile=solarized --title="工作" --active
--tab --profile=solarized --title="tmux"
--tab --profile=solarized --title="docu"
--tab -vv --profile=lee --title="执行" 2>>/tmp/mgt.log 2>&1

看这里，我按照通俗一点的理解你的意思是不是类似于，执行以上6个标签命令，希望执行后在终端可以同时打开6个新标签并且是按照你所出入的留个命令执行的标签页

**Reply by ahvlee** [reply]
我的意思是执行终端的时候自动开几个标签页。每个标签页有自己的标题《可以从bash按排的》，以后如果要的话，在每个标签页用-e自动的执行别的命令。
你没有gnome-terminal吗？

**Reply by ArchieMeng** [reply]
有点奇怪的需求。可以确定的是，deepin-terminal大概可以实现启动的时候开多个tab,但无法指定标题。因为deepin-terminal的标题是根据终端运行的命令来确定的。

**Reply by BLumia** [reply]
终端的标题应该是可被用户覆盖的，有个 `TAB_CUSTOM_NAME_PROPERTY` 来着。

**Comment by meiyixiang** [comment]
不好意思，我不是研发很少会用到终端。

**Reply by ahvlee** [reply]
明白了。我是每时每刻用终端的。
